protocols/v2/subprotocols/common-messages/src/setup_connection.rs

```diff
@@ -84,8 +84,23 @@ impl<'decoder> SetupConnection<'decoder> {
 
                 work_selection && version_rolling
             }
-            // TODO
-            _ => todo!(),
+            Protocol::JobDeclarationProtocol => {
+                let available = available_flags.reverse_bits();
+                let required = required_flags.reverse_bits();
+
+                let requires_async_job_mining_passed = (required >> 31) & 1 > 0;
+                let requires_async_job_mining_self = (available >> 31) & 1 > 0;
+
+                let specific_flags_check =
+                    !requires_async_job_mining_self || requires_async_job_mining_passed;
+                let general_flags_check = (available & required) == required;
```

Comment:
I don't understand semantic of this.
You just have to check the 31st bit, why you do the & bitwise between the whole `available` and `required`?

Reply:
I agree this line is unnecessary, and should be replaced by the `match` suggestion above
```suggestion
```